include/nbl/core/IBuffer.h

```diff
@@ -19,6 +20,29 @@ class IBuffer : public virtual IReferenceCounted
     public:
         //! size in BYTES
         virtual const uint64_t& getSize() const = 0;
+
+        enum E_USAGE_FLAGS : uint32_t
+        {
+            EUF_TRANSFER_SRC_BIT = 0x00000001,
+            EUF_TRANSFER_DST_BIT = 0x00000002,
+            EUF_UNIFORM_TEXEL_BUFFER_BIT = 0x00000004,
+            EUF_STORAGE_TEXEL_BUFFER_BIT = 0x00000008,
+            EUF_UNIFORM_BUFFER_BIT = 0x00000010,
+            EUF_STORAGE_BUFFER_BIT = 0x00000020,
+            EUF_INDEX_BUFFER_BIT = 0x00000040,
+            EUF_VERTEX_BUFFER_BIT = 0x00000080,
+            EUF_INDIRECT_BUFFER_BIT = 0x00000100
+        };
+
+        struct SCreationParams
+        {
+            uint64_t size;
+            E_USAGE_FLAGS usage;
+            asset::E_SHARING_MODE sharingMode;
+            uint32_t queueFamilyIndexCount;
+            const uint32_t* queuueFamilyIndices;

```

Comment:
this stuff shouldn't go in `ICPUBuffer` so it shouldn't be in the common base of `IBuffer`

Also there's two `IBuffer`s, a `core` one and an `asset` one.

All this extra stuff should go into `asset::IBuffer`

Reply:
Ah right. I was wondering why `IBuffer` was in `nbl::core` while `IImage` was in `nbl::asset`.